lib/Sema/Constraint.cpp

```diff
@@ -784,7 +784,7 @@ Constraint *Constraint::create(ConstraintSystem &cs, ConstraintKind kind,
   assert((kind != ConstraintKind::ConformsTo &&
           kind != ConstraintKind::NonisolatedConformsTo &&
           kind != ConstraintKind::TransitivelyConformsTo) ||
-         second->isExistentialType());
+         second->isAnyExistentialType());
```

Comment:
Should we unwrap the metatype before we generate the constraint though, in the caller?

Reply:
I don't think we can since we rely on `matchExistentialTypes` to iteratively match each level of metatype, i.e if you have `$T0 <> P.Type`, we need to form an unsolved constraint with `P.Type` since we wouldn't want it to succeed if `$T0` ends up being simply `S` rather than `S.Type`.